builder/dockerfile/internals_test.go

```diff
@@ -63,21 +65,13 @@ func readAndCheckDockerfile(t *testing.T, testName, contextDir, dockerfilePath,
 		}
 	}()
 
-	context, err := builder.MakeTarSumContext(tarStream)
-	require.NoError(t, err)
-
-	defer func() {
-		if err = context.Close(); err != nil {
-			t.Fatalf("Error when closing tar context: %s", err)
-		}
-	}()
-
-	options := &types.ImageBuildOptions{
-		Dockerfile: dockerfilePath,
+	if dockerfilePath == "" { // handled in BuildWithContext
+		dockerfilePath = builder.DefaultDockerfileName
 	}
 
-	b := &Builder{options: options, context: context}
-
-	_, err = b.readAndParseDockerfile()
+	remote, df, err := remotecontext.Detect(context.Background(), "", dockerfilePath, tarStream, nil)
+	if err == nil {
+		_, err = NewBuilder(context.Background(), &types.ImageBuildOptions{Dockerfile: dockerfilePath}, nil, remote, df)
```

Comment:
With the suggestion above this could just call `readAndParseDockerfile()` directly I think.